plugins/modules/s3_bucket.py

```diff
@@ -163,6 +163,12 @@
     type: bool
     default: false
     version_added: 6.0.0
+  object_lock_enabled:
+    description:
+      - Whether S3 Object Lock to be enabled
+    type: bool
+    default: false
+    version_added: 6.0.0
```

Comment:
Is there any reason why we shouldn't include this feature in 5.3.0? @tremble

Reply:
Nothing I can see.  However, I wouldn't be surprised if it breaks compatibility with one of the many "S3 compatible" APIs.